### PR TITLE
Implement dds_flavors_v3 data source

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
Implement DDS flavors data source

## PR Checklist

* [x] Refers to: #619 #717 
* [x] Tests added/passed.
* [x] Documentation added.

## Acceptance Steps Performed

```
=== RUN   TestAccDDSFlavorV3DataSource_basic
--- PASS: TestAccDDSFlavorV3DataSource_basic (38.33s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			"az_status": item.AZStatus,
+		}
+		flavorList = append(flavorList, flavor)
+	}
+
+	if len(flavorList) < 1 {
+		return fmt.Errorf("your query returned no results. Please change your search criteria and try again")
+	}
+
+	d.SetId("flavors")
```
Comment (outcatcher) [inline opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go:119]: Is ID is always the same? Won't there be a problem if several data source instances are used?

File: opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	mErr := multierror.Append(nil,
+		d.Set("flavors", flavorList),
+		d.Set("region", GetRegion(d, config)),
+	)
+	return mErr.ErrorOrNil()
+}
+
+func filterFlavor(item flavors.Flavor, flavorType, vCpus, memory string) bool {
```
Comment (outcatcher) [inline opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go]: Better to rename it to something like `matchesFilters`

File: opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	if err != nil {
+		return fmt.Errorf("unable to all flavor pages: %s", err)
+	}
+
+	flavorsList, err := flavors.ExtractFlavors(pages)
+	if err != nil {
+		return fmt.Errorf("unable to extract flavors: %s", err)
+	}
+
+	flavorList := make([]map[string]interface{}, 0)
```
Comment (outcatcher) [inline opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go]: `flavorList` and `flavorsList` are too similar, please rename variables to make it easier to distinguish between them

File: opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	filterType := d.Get("type").(string)
```
Comment (outcatcher) [inline opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3.go]: Maybe name it something like `expectedType`, `expectedVCPUs`, etc.?

File: opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3_test.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+func testAccCheckDDSFlavorV3DataSourceID(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("can't find DDS Flavor data source: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
```
Comment (outcatcher) [inline opentelekomcloud/data_source_opentelekomcloud_dds_flavors_v3_test.go:34]: This is a trivial case, as ID is always set to `flavors`

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**